config.m4

```diff
@@ -77,9 +77,14 @@ if test "$PHP_MONGODB" != "no"; then
                  [no])
 
   if test "$PHP_COVERAGE" = "yes"; then
-      PHP_CHECK_GCC_ARG(-fprofile-arcs,                     COVERAGE_CFLAGS="$COVERAGE_CFLAGS -fprofile-arcs")
-      PHP_CHECK_GCC_ARG(-ftest-coverage,                    COVERAGE_CFLAGS="$COVERAGE_CFLAGS -ftest-coverage")
-      EXTRA_LDFLAGS="$COVERAGE_CFLAGS"
+      COVERAGE_CFLAGS="--coverage -g"
+      COVERAGE_LDFLAGS="--coverage"
+
+      if test "$ext_shared" = "yes"; then
+        MONGODB_SHARED_LIBADD="$MONGODB_SHARED_LIBADD $COVERAGE_LDFLAGS"
+      else
+        EXTRA_LDFLAGS="$EXTRA_LDFLAGS $COVERAGE_LDFLAGS"
```

Comment:
I'm not sure that code coverage makes any sense for static builds, so I'm amenable to deleting this `else` case.

In that case, we can also move `PHP_ARG_ENABLE([coverage], ...)` into a conditional so it doesn't even show up for static builds. Let me know if that's also desirable and I can create a second ticket/commit with the change.

Reply:
It would still show up I believe, as that's m4 talk which is done in the phpize stage, whereas the conditionals are done during the configure stage.

IMO it should show up, and say that we don't support it for static builds. If people select it, we should error out in this else case.